### PR TITLE
fix(challenge): Prevent use of String.repeat() in Basic Algorithm

### LINKED ISSUE #4250: Bonfire allows use of str.prototype.repeat.
Challenge [Bonfire: Repeat a string repeat a string](http://www.freecodecamp.com/challenges/bonfire-repeat-a-string-repeat-a-string#?solution=function%20repeat%28str%2C%20num%29%20%7B%0A%20%20ans%20%3D%20%22%22%3B%0A%20%20if%28num%20%3C%3D%200%29%7B%0A%20%20%20%20return%20%22%22%3B%0A%20%20%7D%0A%20%20else%7B%0A%20%20%20%20return%20str%20%2B%20repeat%28str%2C%20num%20-%201%29%3B%0A%20%20%7D%0A%7D%0A%0Arepeat%28%22*%22%2C%203%29%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.80 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

I think this could end up being misleading to beginner programmers who don't realize that repeat is a preexisting function in JS that takes a number as its argument.  Is there a way to restrict the use of repeat for this exercise? I think this would be very valuable to have someone create their own repeat function either through loops or recursion. 

My code:

``` javascript

function repeat(str, num) {
  if(num < 0){
    return "";
  }
  return str.repeat(num);
}

```

### PR DESCRIPTION
**Pull request by Greenheart**
#### Pre-Submission Checklist
- [X] Your pull request targets the `staging` branch of freeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
- [X] Tested changes locally.
- [X] Closes currently open issue (replace XXXX with an issue no): Closes #4250

#### Description
Since algorithm challenges aim to help campers practice creating algorithms, we recommended their own solutions over those built into the language. This PR adds a test that prevent the most straight-forward forms of cheating by using `String.repeat()`.

It won't catch all possible ways, but it's better than nothing.

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json [hunk 1]
```diff
@@ -392,7 +392,8 @@
         "assert(repeatStringNumTimes(\"abc\", 4) === \"abcabcabcabc\", 'message: <code>repeatStringNumTimes(\"abc\", 4)</code> should return <code>\"abcabcabcabc\"</code>.');",
         "assert(repeatStringNumTimes(\"abc\", 1) === \"abc\", 'message: <code>repeatStringNumTimes(\"abc\", 1)</code> should return <code>\"abc\"</code>.');",
         "assert(repeatStringNumTimes(\"*\", 8) === \"********\", 'message: <code>repeatStringNumTimes(\"*\", 8)</code> should return <code>\"********\"</code>.');",
-        "assert(repeatStringNumTimes(\"abc\", -2) === \"\", 'message: <code>repeatStringNumTimes(\"abc\", -2)</code> should return <code>\"\"</code>.');"
+        "assert(repeatStringNumTimes(\"abc\", -2) === \"\", 'message: <code>repeatStringNumTimes(\"abc\", -2)</code> should return <code>\"\"</code>.');",
+        "assert(!/[.]repeat/g.test(code), 'message: The built-in <code>repeat()</code>-method should not be used');"
```
Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json]:
```diff
- [.]
+ \\.
```

**Review by systimotic — CHANGES_REQUESTED** [review]
I don't know if we ever decided whether to use `regex.test(code)` or `code.match(regex)`. Not a dealbreaker for this PR though, I think.

Thanks for picking up this ancient issue!

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json (hunk 1 above)
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json]: Thanks, I tried `\.` but couldn't get it to work. Now I see... ^^

**Comment by Greenheart** [comment]
@systimotic Thanks for the feedback, and pointing me to this issue!

I did a quick check and it seems like there's an even distribution between the two methods. I prefer `RegExp.test()` since it's returning a boolean and we don't really need the actual matching strings when writing simple assertions :blush:

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json (hunk 1 above)
Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json]: Yeah, double escaping gets confusing

**Review by systimotic — APPROVED** [review]
# LGTM 👍 
Thanks @Greenheart!

**Pull request merged**